patches/server/0255-Improve-death-events.patch

```diff
@@ -220,10 +220,31 @@ index 7c0e1963383a1a2862930cf77844e5c8c80e70e3..4c1992bcffd629ef53f14c5a0146eab2
      // CraftBukkit start
      public int getExpReward() {
 diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
-index 984a33d5e1f790a9c78ba57f2dc21fb072a44b3d..dbf442e9686e59723ed0456f97e472cc663f8cc7 100644
+index 984a33d5e1f790a9c78ba57f2dc21fb072a44b3d..5524a26f6401731872b11c0360fc6eb6cc2e042a 100644
 --- a/src/main/java/net/minecraft/world/entity/Mob.java
 +++ b/src/main/java/net/minecraft/world/entity/Mob.java
-@@ -1065,7 +1065,13 @@ public abstract class Mob extends LivingEntity implements Targeting {
+@@ -1047,6 +1047,12 @@ public abstract class Mob extends LivingEntity implements Targeting {
+ 
+     }
+ 
++    // Paper start
++    protected boolean shouldLootFrom(EquipmentSlot slot) { // method to avoid to fallback into the global mob loot logic (i.e fox)
```

Comment:
I think this boolean should be inverted the whole way through. To avoid a double flip of the value that just adds confusing. It should be `shouldSkipLoot`, and then adjust everything else accordingly.